packages/react/scroll-area/src/ScrollArea.tsx

```diff
@@ -186,2234 +80,886 @@ type ScrollAreaPrimitive = Polymorphic.ForwardRefComponent<
+            // prevent window scroll when wheeling on scrollbar
+            if (scrollingWithinScrollbarBounds(scrollPos, maxScrollPos)) {
+              event.preventDefault();
+            }
+          }
+        }}
+        onResize={() => {
+          if (ref.current && context.viewport && computedStyle) {
+            onSizesChange({
+              content: context.viewport.scrollHeight,
+              viewport: context.viewport.offsetHeight,
+              scrollbar: {
+                size: ref.current.clientHeight,
+                paddingStart: toInt(computedStyle.paddingTop),
+                paddingEnd: toInt(computedStyle.paddingBottom),
+              },
+            });
+          }
         }}
-        onPointerDown={handlePointerDown}
-        onPointerUp={handlePointerUp}
-        onPointerMove={handlePointerMove}
-        onWheel={handleWheel}
       />
-    </ScrollbarProvider>
+    </ScrollbarAxisProvider>
   );
-}) as ScrollAreaScrollbarPrimitive;
+}) as ScrollAreaScrollbarYPrimitive;
 
-type ScrollAreaScrollbarXOwnProps = Polymorphic.OwnProps<typeof Primitive>;
-type ScrollAreaScrollbarXPrimitive = Polymorphic.ForwardRefComponent<
+/* -----------------------------------------------------------------------------------------------*/
+
+type ScrollAreaScrollbarImplOwnProps = Polymorphic.Merge<
+  Polymorphic.OwnProps<typeof Primitive>,
+  {
+    onWheelScroll(event: WheelEvent): void;
+    onDragScroll(pointerPosition: { x: number; y: number }): void;
+    onResize(): void;
+  }
+>;
+
+type ScrollAreaScrollbarImplPrimitive = Polymorphic.ForwardRefComponent<
   Polymorphic.IntrinsicElement<typeof Primitive>,
-  ScrollAreaScrollbarXOwnProps
+  ScrollAreaScrollbarImplOwnProps
 >;
 
-const ScrollAreaScrollbarX = React.forwardRef(function ScrollAreaScrollbarX(props, forwardedRef) {
-  const { domSizes } = useScrollAreaStateContext();
-  return useNativeScrollArea() ? null : (
-    <ScrollAreaScrollbar
-      {...props}
-      ref={forwardedRef}
-      axis="x"
-      name={SCROLLBAR_X_NAME}
-      style={{
-        ...props.style,
-        // @ts-ignore
-        [SCROLL_AREA_CSS_PROPS.scrollbarXSize]: domSizes.scrollbarX.height
-          ? domSizes.scrollbarX.height + 'px'
-          : 0,
-      }}
-    />
-  );
-}) as ScrollAreaScrollbarXPrimitive;
+const ScrollAreaScrollbarImpl = React.forwardRef((props, forwardedRef) => {
+  const { onDragScroll, onWheelScroll, onResize, ...scrollbarProps } = props;
+  const context = useScrollAreaContext(SCROLLBAR_NAME);
+  const [scrollbar, setScrollbar] = React.useState<ScrollbarElement | null>(null);
+  const composeRefs = useComposedRefs(forwardedRef, (node) => setScrollbar(node));
+  const rectRef = React.useRef<ClientRect | null>(null);
+  const prevWebkitUserSelectRef = React.useRef<string>('');
+  const viewport = context.viewport;
+  const handleWheelScroll = useCallbackRef(onWheelScroll);
+  const debounceResize = useDebounceCallback(onResize, 10);
```

Comment:
This particular one makes sense to rename because it is calling an `onResize` prop but some of the others aren't. They're being called directly within a scroll listener and I thought it looked weird to see `const handleScroll = () => handleScrollEnd()`... The `debounceScrollEnd` made sense there as it communicated it wasn't immediately firing scroll end. Thoughts?

Reply:
I see, makes sense. In this particular one, is it needed at all? considering you're already debouncing inside `useResizeObserver`?

Reply:
Yes, this one is different, it's to prevent two resizes happening when content and viewport both resize. We only want one `onResize` called once even if they both resize.